packages/ui/src/theme/tokens/colors.ts

```diff
@@ -1,125 +1,101 @@
-  interactive: DesignTokenType;
-  hover: DesignTokenType;
-  focus: DesignTokenType;
-  active: DesignTokenType;
-  disabled: DesignTokenType;
-} & OrdinalScale<DesignTokenType> &
+type OverlayKey = 10 | 20 | 30 | 40 | 50 | 60 | 70 | 80 | 90;
+type OverlayColors<DesignTokenType> = Record<OverlayKey, DesignTokenType>;
+
+type FontVariant =
+  | 'inverse'
+  | 'interactive'
+  | 'hover'
+  | 'focus'
+  | 'active'
+  | 'disabled';
+
+type FontColors<DesignTokenType> = Record<FontVariant, DesignTokenType> &
+  OrdinalScale<DesignTokenType> &
   OrdinalVariation<DesignTokenType>;
 
-type BackgroundColors<DesignTokenType = DesignToken<ColorValue>> = {
+type BackgroundColors<DesignTokenType> = {
   disabled: DesignTokenType;
 } & OrdinalScale<DesignTokenType> &
   OrdinalVariation<DesignTokenType>;
 
-type BorderColors<DesignTokenType = DesignToken<ColorValue>> = {
+type BorderColors<DesignTokenType> = {
   disabled: DesignTokenType;
   pressed: DesignTokenType;
   focus: DesignTokenType;
   error: DesignTokenType;
 } & OrdinalScale<DesignTokenType>;
 
-type ColorTypes<DesignTokenType = DesignToken<ColorValue>> =
-  | { [key in ScaleKeys]: DesignTokenType }
-  | FontColors
-  | BackgroundColors
-  | DesignTokenType
-  | BorderColors;
-
-type WebColorTypes = ColorTypes<WebDesignToken<ColorValue>>;
-
-export type Colors = {
-  // base color palette
-  red: ColorScale;
-  orange: ColorScale;
-  yellow: ColorScale;
-  green: ColorScale;
-  teal: ColorScale;
-  blue: ColorScale;
-  purple: ColorScale;
-  pink: ColorScale;
-  neutral: ColorScale;
-  white: DesignToken<ColorValue>;
-  black: DesignToken<ColorValue>;
-
-  // Semantic colors
-  font: FontColors;
-  background: BackgroundColors;
-  border: BorderColors;
-  brand: {
-    primary: ColorScale;
-    secondary: ColorScale;
-  };
-
-  overlay: OverlayColors;
-
-  [key: string]: ColorTypes | Record<string, ColorTypes>;
+type ColorTypes<DesignTokenType> =
+  | Record<ScaleKey, DesignTokenType>
+  | FontColors<DesignTokenType>
+  | BackgroundColors<DesignTokenType>
+  | BorderColors<DesignTokenType>
+  | DesignTokenType;
+
+type PaletteColor =
+  | 'red'
+  | 'orange'
+  | 'yellow'
+  | 'green'
+  | 'teal'
+  | 'blue'
+  | 'purple'
+  | 'pink'
+  | 'neutral';
+
+// base color palette
+type BasePalette<DesignTokenType> = Record<
+  PaletteColor,
+  ColorScale<DesignTokenType>
+>;
+
+export type Colors<DesignTokenType = DesignToken<ColorValue> | string> =
+  BasePalette<DesignTokenType> & {
+    white: DesignTokenType;
+    black: DesignTokenType;
+
+    // Semantic colors
+    font: FontColors<DesignTokenType>;
+    background: BackgroundColors<DesignTokenType>;
+    border: BorderColors<DesignTokenType>;
+    brand: {
+      primary: ColorScale<DesignTokenType>;
+      secondary: ColorScale<DesignTokenType>;
+    };
+
+    overlay: OverlayColors<DesignTokenType>;
+  } & Record<
+      string,
+      ColorTypes<DesignTokenType> | Record<string, ColorTypes<DesignTokenType>>
+    >;
```

Comment:
Will this allow users to add colors that are not in the theme? That has been a request

Reply:
I _believe_ so, this is `[key: string]: WebColorTypes | Record<string, WebColorTypes>` from [what was line 97](https://github.com/aws-amplify/amplify-ui/pull/2902/files#diff-f3f72291b73e9cbe2298b56eec3eb54efb801d7c2d7086366d905563475fe08dL97). Will look in to it

Reply:
Added a unit test asserting it: https://github.com/aws-amplify/amplify-ui/pull/2902/commits/95340ffa93bef598c96a01b7e426ec508231fb9d